src/ensembl/utils/plugin.py

```diff
@@ -63,6 +64,23 @@ def pytest_addoption(parser: Parser) -> None:
     )
 
 
+def pytest_configure(config: Config) -> None:
+    """Allows plugins and conftest files to perform initial configuration.
+
+    More information: https://docs.pytest.org/en/latest/reference/reference.html#std-hook-pytest_configure
+
+    Args:
+        config: The pytest config object.
+
+    """
+    # Load server information
+    server_url = make_url(config.getoption("server"))
+    # If password set, treat it as an environment variable that needs to be resolved
+    if server_url.password:
+        server_url = server_url.set(password=os.path.expandvars(server_url.password))
+        config.option.server = str(server_url)
```

Comment:
This doesn't work, because the password is changed to `***` during conversion of the URL to `str`.

The following change would make it work:
```suggestion
        config.option.server = server_url.render_as_string(hide_password=False)
```